### ISSUE TITLE
EMSG boxes are not being completely parsed

### ISSUE THREAD

**Issue by chrisfillmore** [google/shaka-player#1340]
**Have you read the [FAQ](https://goo.gl/JE1Sy5) and checked for duplicate open issues?**:
Yes

**What version of Shaka Player are you using?**:
2.3.3

**Can you reproduce the issue with our latest release version?**:
Yes

**Can you reproduce the issue with the latest code from `master`?**:
Did not check

**Are you using the demo app or your own custom app?**:
Custom app

**If custom app, can you reproduce the issue using our demo app?**:
Unknown. Does any demo content contain emsg boxes?

**What browser and OS are you using?**:
Chrome on macOS

**What are the manifest and license server URIs?**:
I could potentially provide a demo app to reproduce, if required. But I think you can follow the thread through the code to see the issue.

**What did you do?**
Attempt to play one of our live DAI streams which contains EMSG boxes.

**What did you expect to happen?**
All EMSG boxes should get parsed and published.

**What actually happened?**
We only see one EMSG box published (out of three, I believe) per segment.

We just upgraded from 2.2.8 to 2.3.3. We now have a problem with emsg boxes not being parsed/published. It looks like the bug was introduced in https://github.com/google/shaka-player/commit/a93455a917e0c97fd4a57e8bb52d25659fc5ced8. Specifically, `StreamingEngine#parseEMSG_` now does:

```
box.parser.stop();
```

...and `Mp4Parser#parse` now checks if `!this.done_` while parsing:

```
while (reader.hasMoreData() && !this.done_) {
  this.parseNext(0, reader, opt_partialOkay);
}
```

Since `parseEMSG_` is passed to `Mp4Parser#fullBox` as the `definition`, and this is invoked each time `Mp4Parser#parseNext` is called, the parser stops parsing after the first emsg box.

This was introduced in v2.3.0 but I don't see mention of #1106 in the release notes.

**Comment by chrisfillmore** [comment]
Thanks @TheModMaker!

**Comment by joeyparrish** [comment]
Fix cherry-picked for v2.3.4.
